synapse/storage/databases/main/group_server.py

```diff
@@ -528,15 +546,15 @@ def _get_all_groups_for_user_txn(txn):
             "get_all_groups_for_user", _get_all_groups_for_user_txn
         )
 
-    async def get_groups_changes_for_user(self, user_id, from_token, to_token):
+    async def get_groups_changes_for_user(self, user_id: str, from_token, to_token):
```

Comment:
I think this is:

```suggestion
    async def get_groups_changes_for_user(self, user_id: str, from_token: int, to_token: int) -> List[JsonDict]:
```

(And the same return type on the inner function.)